### PR TITLE
Add manifest translate command

### PR DESCRIPTION
**Pull request by howardjohn**
This implements a new `manifest translate` command. 

Currently, it will take each chart, and give you the new `values.yaml` to apply to it, the steps to take-over existing resources, and the command to install with helm. Users can use any or all of these depending on their needs.

Additionally, it will warn if there is a difference in result. This could be from unsupported field usage or otherwise.

Additionally, it will warn if we cannot migrate. This applies for charts that don't exist in helm release (gateway).

Future steps:
* Give more guidance on gateway cases.
* Give more guidance on `diff` cases.
* Clean up outputted values such that we do not write info about other components, or other cruft we don't need.

----

Example output:

Folder contains `base-values.yaml  install-base.sh*  install-pilot.sh*  pilot-values.yaml  README.md`

readme contents:
---

# IstioOperator to Helm Migration

This folder contains auto-generated output from the `istioctl manifest translate` command.
Note the `manifest translate` command only outputs this folders contents, and does not modify the cluster state.

Follow the instructions below for each component to complete the migration.

# Components
* ✅ **Component `base`**: migration is supported!

  The translated values have been written to base-values.yaml.
  You may use these directly, or follow the guided `install-base.sh` script.
* ✅ **Component `pilot`**: migration is supported!

  The translated values have been written to pilot-values.yaml.
  You may use these directly, or follow the guided `install-pilot.sh` script.
* ❌ **Component `istio-ingressgateway`**: migration is **NOT** directly supported!

### CODE REVIEW

**Review by bleggett — APPROVED** [review]
This broadly makes sense to me - I still think we need to actually group our charts more logically and simply according to suggested upgrade considerations into e.g. (`istio-dataplane` and `istio-controlplane`) which would make this overall process a bit less fiddly for people - but we gotta start somewhere and this is the start of a way out of `IstioOperator` land, which is a good thing in my book.

File: operator/cmd/mesh/manifest-translate.go [hunk 1]
```diff
@@ -0,0 +1,196 @@
...
+			helmName := fmt.Sprintf("diff-%s-helm-output.yaml", name)
+			istioctlName := fmt.Sprintf("diff-%s-istioctl-output.yaml", name)
+			if err := write(helmName, helmManifests); err != nil {
+				return err
+			}
+			if err := write(istioctlName, istioctlManifests); err != nil {
+				return err
+			}
+			diffWarn = fmt.Sprintf(`
+  ⚠️ Component rendering is different between Istioctl and Helm!
```
[the remark this answers is not in the document]
Comment (howardjohn) [reply operator/cmd/mesh/manifest-translate.go:172]: The primary expectation is when they use anything under `components.k8s`. Try with `manifest translate -f operator/cmd/mesh/testdata/manifest-generate/input/pilot_k8s_settings.yaml` for an example

[the remark this answers is not in the document]
Comment (howardjohn) [reply operator/cmd/mesh/manifest-translate.go:172]: Or phrase differently: anytime `postProcess` does something

File: operator/cmd/mesh/manifest-translate.go [hunk 2]
```diff
@@ -0,0 +1,196 @@
...
+			continue
+		}
+		commands := []string{"#!/usr/bin/env bash", "", "# Label/Annotate resources to mark them a part of the Helm release."}
+		ns := info.ComponentSpec.Namespace
+		vals, _ := info.Values.GetPathMap("spec.values")
+		valuesName := fmt.Sprintf("%v-values.yaml", name)
+		if err := write(valuesName, vals.YAML()); err != nil {
+			return err
+		}
+		for _, m := range info.Manifest {
```
[the remark this answers is not in the document]
Comment (howardjohn) [reply operator/cmd/mesh/manifest-translate.go:141]: `migrate-to-helm` seems a nice name. We can also split the one-time steps (annotate) vs the continual steps  (helm upgrade)?

[the remark this answers is not in the document]
Comment (bleggett) [reply operator/cmd/mesh/manifest-translate.go:141]: Either or both of those seems fine to me. Slight preference for a split, yeah (command or flag).

-  `manifest translate` and `manifest translate --migrate`
- `manifest migrate-helm` and `manifest migrate-helm --convert/translate` 
- or just `manifest translate` and `manifest migrate`.

probably 2  users of this 

1. People that wanna quickly convert one set of values to the other for copypaste purposes (docs, slack, other things) (dump to stdout/file)
2. People that wanna migrate a live cluster (migrate with scripts)

File: operator/cmd/mesh/manifest.go [hunk 3]
```diff
@@ -29,19 +29,24 @@ func ManifestCmd(ctx cli.Context) *cobra.Command {
...
 
 	args := &RootArgs{}
 
 	mgc := ManifestGenerateCmd(ctx, args, mgcArgs)
+	mtc := ManifestTranslateCmd(ctx, mtcArgs)
 	ic := InstallCmd(ctx)
 
 	addFlags(mc, args)
 	addFlags(mgc, args)
+	addFlags(mtc, args)
```
Comment (hanxiaop) [inline operator/cmd/mesh/manifest.go:42]: The `dry-run` flag is added but not used.

Comment (howardjohn) [reply operator/cmd/mesh/manifest.go:42]: thanks

**Review by hanxiaop — APPROVED** [review]

**Pull request merged**